src/kbmod/results.py

```diff
@@ -0,0 +1,591 @@
+"""Results is a column-based data structure for tracking results with additional global data
+and helper functions for filtering and maintaining consistency between different attributes in each row.
+"""
+
+import numpy as np
+import os.path as ospath
+from pathlib import Path
+
+from astropy.table import Table, vstack
+
+from kbmod.file_utils import FileUtils
+from kbmod.trajectory_utils import make_trajectory, trajectory_from_np_object
+from kbmod.search import Trajectory
+
+
+class Results:
+    """This class stores a collection of related data from all of the kbmod results.
+
+    At a minimum it contains columns for the trajectory information:
+    (x, y, vx, vy, likelihood, flux, obs_count)
+    but additional columns can be added as needed.
+
+    Attributes
+    ----------
+    table : `astropy.table.Table`
+        The stored results data.
+    track_filtered : `bool`
+        Whether to track (save) the filtered trajectories. This will use
+        more memory and is recommended only for analysis.
+    filtered : `dict`
+        A dictionary mapping a string of filtered name to a table
+        of the results removed by that filter.
+    """
+
+    # The required columns list gives a list of tuples containing
+    # (column name, dype, default value) for each required column.
+    _required_cols = [
+        ("x", "int64", 0),
+        ("y", "int64", 0),
+        ("vx", "float64", 0.0),
+        ("vy", "float64", 0.0),
+        ("likelihood", "float64", 0.0),
+        ("flux", "float64", 0.0),
+        ("obs_count", "int64", 0),
+    ]
```

Comment:
Changed. Although I needed to use `float`, `int`, etc because the numpy versions are deprecated in favor of those.

Reply:
Interesting, the recommendation is to insist on using `np.float64` and `np.float32` instead of system default alias `np.float`. Good to know, thanks:

https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated